### PR TITLE
[ACCOUNT]: edit account details

### PR DESCRIPTION
**Pull request by AnaSireag**

### CODE REVIEW

File: src/main/java/com/bootcamp/demo/Controller/AccountController.java [hunk 1]
```diff
@@ -31,6 +31,30 @@ void add(@RequestBody Account account) throws ControllerException {
...
+                        accountDetails.getPhoneNumber(),
+                        accountDetails.getAddress(),
+                        accountDetails.getDateOfBirth()
+                );
+                this.service.update(account);
+            }
+
+        } catch (ServiceException | ExecutionException | InterruptedException e) {
+            throw new ControllerException(e.getMessage());
+        }
```
Comment (cosmin-195) [inline src/main/java/com/bootcamp/demo/Controller/AccountController.java:55]: This exception isn't caught anywhere, so I think you should return a response body instead, along with an appropriate http status.

File: src/main/java/com/bootcamp/demo/business/AccountService.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            account.setPassword(encryptor.encryptSHA256(account.getPassword()));
+            System.out.println(account.getPassword());
+            return repository.add(account);
+        } catch (ExecutionException | InterruptedException exception) {
+            throw new ServiceException(exception.getMessage());
+        }
+    }
+
+    @Override
+    public Account findOne(String username) throws ServiceException, ExecutionException, InterruptedException {
```
Comment (andreibacer) [inline src/main/java/com/bootcamp/demo/business/AccountService.java:39]: `retrieve` might be a more suitable name for this method.

File: src/main/java/com/bootcamp/demo/business/AccountService.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        return repository.findOne(username);
+    }
+
+    @Override
+    public String update(Account account) throws ServiceException, ExecutionException, InterruptedException {
+        System.out.println("Service --> " + account);
```
Comment (andreibacer) [inline src/main/java/com/bootcamp/demo/business/AccountService.java:45]: Logging should be more explicit

File: src/main/java/com/bootcamp/demo/data_access/AccountRepository.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        // return optional?
+        Firestore db = FirestoreClient.getFirestore();
+        CollectionReference collectionReference = db.collection("accounts");
+        DocumentReference documentReference = collectionReference.document(username);
+        ApiFuture<DocumentSnapshot> future = documentReference.get();
+        DocumentSnapshot documentSnapshot = future.get();
+//        if(documentSnapshot.exists())
+//            return documentSnapshot.toObject(Account.class);
+        if (documentSnapshot.exists()) {
+            Map<String, Object> data = documentSnapshot.getData();
```
Comment (andreibacer) [inline src/main/java/com/bootcamp/demo/data_access/AccountRepository.java:28]: Can't we map the document to an Account without extracting properties one by one?

File: src/main/java/com/bootcamp/demo/data_access/AccountRepository.java [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+@Repository
+public class AccountRepository implements Repo<Account> {
+    @Override
+    public Account findOne(String username) throws ExecutionException, InterruptedException {
+        // return optional?
```
Comment (andreibacer) [inline src/main/java/com/bootcamp/demo/data_access/AccountRepository.java:19]: an Optional might be a better idea, especially if we are using `find` in method name.

File: src/main/java/com/bootcamp/demo/data_access/AccountRepository.java [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+        Firestore db = FirestoreClient.getFirestore();
+        DocumentReference docRef = db.collection("accounts").document(account.getUsername());
+        ApiFuture<WriteResult> writeResult = docRef.set(account);
+        return writeResult.get().getUpdateTime().toString();
+    }
+
+    @Override
+    public String update(Account account) throws ExecutionException, InterruptedException {
+        Firestore db = FirestoreClient.getFirestore();
+        CollectionReference collectionReference = db.collection("accounts");
```
Comment (andreibacer) [inline src/main/java/com/bootcamp/demo/data_access/AccountRepository.java:52]: inline variable if it is used only once. Also it seems that the code in this method is very similar with the one in the methods above. Can't the common code be extracted and reused where needed?

File: src/main/java/com/bootcamp/demo/data_access/AccountRepository.java [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+        return false;
+    }
+
+    public List<Account> getAll() throws ExecutionException, InterruptedException {
+        Firestore db = FirestoreClient.getFirestore();
+        ApiFuture<QuerySnapshot> query = db.collection("accounts").get();
+        QuerySnapshot querySnapshot = query.get();
+        List<QueryDocumentSnapshot> documents = querySnapshot.getDocuments();
+        List<Account> accountList = new ArrayList<>();
+        for (QueryDocumentSnapshot q : documents) {
```
Comment (andreibacer) [inline src/main/java/com/bootcamp/demo/data_access/AccountRepository.java:84]: try to rewrite this with Java 8. Something like:

`documents.stream()
                     .filter(QueryDocumentSnapshot::exists)
                     .map(this::createAccount)
                     .collect(Collectors.toList())`

**Pull request closed without merging**